File: client/src/features/roles/dto/response/role.response.ts
import type { UserBriefResponse } from '../../../../shared/dto/response'

export type RoleStatus = 'ACTIVE' | 'INACTIVE'

export interface RoleResponse {
  id: string
  name: string
  status: RoleStatus
  createdBy: UserBriefResponse
  updatedBy: UserBriefResponse
  createdAt: string
  updatedAt: string
}

export type RoleScopeType = 'PRODUCT' | 'SERVICE'
export type PermissionType = 'ALL' | 'READ' | 'READ_AND_WRITE'
export type RoleScopeStatus = 'ACTIVE' | 'INACTIVE'

export interface RoleScopeResponse {
  id: string
  scopeType: RoleScopeType
  scopeId: string
  scopeName: string
  permission: PermissionType
  status: RoleScopeStatus
}

File: backend/sentinel-api/src/main/java/com/sentinel/api/role/dto/response/RoleResponse.java
package com.sentinel.api.role.dto.response;

import com.sentinel.api.common.dto.response.UserBriefResponse;
import com.sentinel.api.role.entity.RoleStatus;

import java.time.Instant;

public record RoleResponse(
    String id,
    String name,
    RoleStatus status,
    UserBriefResponse createdBy,
    UserBriefResponse updatedBy,
    Instant createdAt,
    Instant updatedAt) {
}
